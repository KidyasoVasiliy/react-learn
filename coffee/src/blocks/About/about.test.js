import React from 'react';
import { About } from './index';
import { shallow } from 'enzyme';

describe('Testing <About />', () => {
  it('About have renderer correctly', () => {
    const about = shallow(<About paragraphs={['lorem', 'lorem2']}/>);
    expect(about).toMatchSnapshot();
  })

  it('About have renderer correctly', () => {
    const about = shallow(<About info={{name: 'lorem'}}/>);
    expect(about).toMatchSnapshot();
  })
});
