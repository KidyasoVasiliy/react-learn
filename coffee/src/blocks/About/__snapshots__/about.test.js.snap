// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <About /> About have renderer correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About
    paragraphs={
      Array [
        "lorem",
        "lorem2",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="about__item"
        >
          <Heading />
        </div>,
        <div
          className="about__item"
        >
          <BeansLogo
            tintColor="black"
          />
        </div>,
        <div
          className="about__item"
        >
          <div
            className="about__item-text"
          >
            <Paragraph
              text="lorem"
            />
          </div>
          <div
            className="about__item-text"
          >
            <Paragraph
              text="lorem2"
            />
          </div>
        </div>,
      ],
      "className": "about",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Heading />,
          "className": "about__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BeansLogo
            tintColor="black"
          />,
          "className": "about__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "tintColor": "black",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <div
                className="about__item-text"
              >
                <Paragraph
                  text="lorem"
                />
              </div>,
              <div
                className="about__item-text"
              >
                <Paragraph
                  text="lorem2"
                />
              </div>,
            ],
            null,
          ],
          "className": "about__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "lorem-0",
            "nodeType": "host",
            "props": Object {
              "children": <Paragraph
                text="lorem"
              />,
              "className": "about__item-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "lorem",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "lorem2-1",
            "nodeType": "host",
            "props": Object {
              "children": <Paragraph
                text="lorem2"
              />,
              "className": "about__item-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "lorem2",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="about__item"
          >
            <Heading />
          </div>,
          <div
            className="about__item"
          >
            <BeansLogo
              tintColor="black"
            />
          </div>,
          <div
            className="about__item"
          >
            <div
              className="about__item-text"
            >
              <Paragraph
                text="lorem"
              />
            </div>
            <div
              className="about__item-text"
            >
              <Paragraph
                text="lorem2"
              />
            </div>
          </div>,
        ],
        "className": "about",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Heading />,
            "className": "about__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BeansLogo
              tintColor="black"
            />,
            "className": "about__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "tintColor": "black",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <div
                  className="about__item-text"
                >
                  <Paragraph
                    text="lorem"
                  />
                </div>,
                <div
                  className="about__item-text"
                >
                  <Paragraph
                    text="lorem2"
                  />
                </div>,
              ],
              null,
            ],
            "className": "about__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "lorem-0",
              "nodeType": "host",
              "props": Object {
                "children": <Paragraph
                  text="lorem"
                />,
                "className": "about__item-text",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "text": "lorem",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "lorem2-1",
              "nodeType": "host",
              "props": Object {
                "children": <Paragraph
                  text="lorem2"
                />,
                "className": "about__item-text",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "text": "lorem2",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing <About /> About have renderer correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About
    info={
      Object {
        "name": "lorem",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="about__item"
        >
          <Heading />
        </div>,
        <div
          className="about__item"
        >
          <BeansLogo
            tintColor="black"
          />
        </div>,
        <div
          className="about__item"
        >
          <div
            className="about__item-info"
          >
            <div
              className="info__item"
            >
              <Paragraph />
            </div>
            <div
              className="info__item"
            >
              <Paragraph />
            </div>
          </div>
        </div>,
      ],
      "className": "about",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Heading />,
          "className": "about__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BeansLogo
            tintColor="black"
          />,
          "className": "about__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "tintColor": "black",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <div
              className="about__item-info"
            >
              <div
                className="info__item"
              >
                <Paragraph />
              </div>
              <div
                className="info__item"
              >
                <Paragraph />
              </div>
            </div>,
          ],
          "className": "about__item",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                <div
                  className="info__item"
                >
                  <Paragraph />
                </div>,
                <div
                  className="info__item"
                >
                  <Paragraph />
                </div>,
              ],
              "className": "about__item-info",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Paragraph />,
                  "className": "info__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "text": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Paragraph />,
                  "className": "info__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "text": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="about__item"
          >
            <Heading />
          </div>,
          <div
            className="about__item"
          >
            <BeansLogo
              tintColor="black"
            />
          </div>,
          <div
            className="about__item"
          >
            <div
              className="about__item-info"
            >
              <div
                className="info__item"
              >
                <Paragraph />
              </div>
              <div
                className="info__item"
              >
                <Paragraph />
              </div>
            </div>
          </div>,
        ],
        "className": "about",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Heading />,
            "className": "about__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BeansLogo
              tintColor="black"
            />,
            "className": "about__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "tintColor": "black",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <div
                className="about__item-info"
              >
                <div
                  className="info__item"
                >
                  <Paragraph />
                </div>
                <div
                  className="info__item"
                >
                  <Paragraph />
                </div>
              </div>,
            ],
            "className": "about__item",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  <div
                    className="info__item"
                  >
                    <Paragraph />
                  </div>,
                  <div
                    className="info__item"
                  >
                    <Paragraph />
                  </div>,
                ],
                "className": "about__item-info",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Paragraph />,
                    "className": "info__item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "text": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Paragraph />,
                    "className": "info__item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "text": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
